benchmark/benchmark_slog_test.go

```diff
@@ -0,0 +1,142 @@
+//go:build go1.21
+// +build go1.21
+
+/*
+Copyright 2021 The logr Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logr
+
+import (
+	"log/slog"
+	"os"
+	"testing"
+
+	"github.com/go-logr/logr"
+)
+
+//
+// slogSink wrapper of discard
+//
```

Comment:
Did you learn anything from these benchmarks? I'm just curious.

Reply:
I learned that funcr with JSON is more than 2x faster than slog's JSON handler, but has more allocs.  And that the slogSink (wrapper) path is (as expected) low overhead.

Reply:
I did a similar comparison between slog's JSON handler and zap (https://github.com/kubernetes/kubernetes/pull/120696/commits/9100c3cb8a38caf70009f596de10853c473f8db0). Slog was slower, so there was no justification for switching from zap to slog as backend in Kubernetes.